h5netcdf/core.py

```diff
@@ -400,6 +444,10 @@ def _create_dim_scales(self):
                 dims = self._variables[dim].dimensions
                 coord_ids = np.array([dim_order[d] for d in dims], 'int32')
                 h5ds.attrs['_Netcdf4Coordinates'] = coord_ids
+            # Might have to be resized if it already exists.
+            elif h5ds.maxshape == (None, ) and \
```

Comment:
I'm confused by the current placement of this clause. I think it belongs about 6 lines up, after `if dim not in self._h5group`.